src/app/services/posts.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+
+
+const baseURL = 'https://jsonplaceholder.typicode.com/';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PostsService {
+  getSinglePost = new Subject<Object>()
+  public posts = new BehaviorSubject<Array<string>>([]);
```

Comment:
you can also type Subject<string[]>, easier to read, and I suppose this is Subject<Post[]>... also, public is unnecessary